config/fcos/v1_6_exp/validate.go

```diff
@@ -51,31 +52,43 @@ func (conf Config) Validate(c path.ContextPath) (r report.Report) {
 }
 
 func (d BootDevice) Validate(c path.ContextPath) (r report.Report) {
-	if d.Layout != nil {
-		switch *d.Layout {
+	var layout = d.Layout
+	if layout != nil {
+		switch *layout {
 		case "aarch64", "ppc64le", "x86_64":
+			if util.IsTrue(d.Luks.Cex.Enabled) {
+				r.AddOnError(c.Append(*layout), common.ErrCexArchitectureMismatch)
```

Comment:
While adding tests for this part, I found out that this code is apparently not correct.
It comes from https://github.com/coreos/butane/pull/484 and apparently we did not add tests for it.
Working on it now.